agent/rpc/middleware/recovery.go

```diff
@@ -0,0 +1,24 @@
+package middleware
```

Comment:
If we want to keep the grpc vs rpc visibility in logs then we should keep them separate

Reply:
We are printing the call stack as part of the error log  we should be able to differentiate gRPC from RPC at that level.

Reply:
Updated here https://github.com/hashicorp/consul/pull/16022/commits/8c3c7dca87a907da21a54fd88e8e855a5ef2af1e
@boxofrad can I also get your 👀 on this since it makes some changes to gRPC middleware 🙏